ESPController/src/HAL_ESP32.cpp

```diff
@@ -77,65 +77,66 @@ uint8_t HAL_ESP32::ReadTCA6408InputRegisters()
+    TCA9534APWR_Value = readByte(I2C_NUM_0, TCA9534APWR_ADDRESS, TCA9534APWR_INPUT);
     return TCA9534APWR_Value & TCA9534APWR_INPUTMASK;
 }
 
 void HAL_ESP32::SetOutputState(uint8_t outputId, RelayState state)
 {
     ESP_LOGD(TAG, "SetOutputState %u=%u", outputId, state);
 
-    //Relays connected to TCA6408A
-    //P4 = RELAY1 (outputId=0)
-    //P5 = RELAY2 (outputId=1)
-    //P6 = RELAY3_SSR (outputId=2)
-    //P7 = EXT_IO_E (outputId=3)
+    // Relays connected to TCA6408A
+    // P4 = RELAY1 (outputId=0)
+    // P5 = RELAY2 (outputId=1)
+    // P6 = RELAY3_SSR (outputId=2)
+    // P7 = EXT_IO_E (outputId=3)
 
     if (outputId <= 3)
     {
         TCA6408_Value = readByte(I2C_NUM_0, TCA6408_ADDRESS, TCA6408_INPUT);
         uint8_t bit = outputId + 4;
         TCA6408_Value = (state == RelayState::RELAY_ON) ? (TCA6408_Value | (1 << bit)) : (TCA6408_Value & ~(1 << bit));
         ESP_ERROR_CHECK_WITHOUT_ABORT(writeByte(I2C_NUM_0, TCA6408_ADDRESS, TCA6408_OUTPUT, TCA6408_Value));
-        //ESP_LOGD(TAG, "TCA6408 reply %i", ret);
-        //TODO: Check return value
+        // ESP_LOGD(TAG, "TCA6408 reply %i", ret);
+        // TODO: Check return value
         TCA6408_Value = readByte(I2C_NUM_0, TCA6408_ADDRESS, TCA6408_INPUT);
     }
 }
 
 void HAL_ESP32::Led(uint8_t bits)
 {
-    //Clear LED pins
+    // Clear LED pins
     TCA9534APWR_Value = TCA9534APWR_Value & B11111000;
-    //Set on
+    // Set on
     TCA9534APWR_Value = TCA9534APWR_Value | (bits & B00000111);
-    //esp_err_t ret =
+    // esp_err_t ret =
     ESP_ERROR_CHECK_WITHOUT_ABORT(writeByte(I2C_NUM_0, TCA9534APWR_ADDRESS, TCA9534APWR_OUTPUT, TCA9534APWR_Value));
 
-    //ESP_LOGD(TAG,"TCA9534 LED reply %i",ret);
-    //TODO: Check return value
+    // ESP_LOGD(TAG,"TCA9534 LED reply %i",ret);
+    // TODO: Check return value
 }
 
 void HAL_ESP32::ConfigureCAN()
 {
-    //Initialize configuration structures using macro initializers
-    can_general_config_t g_config = CAN_GENERAL_CONFIG_DEFAULT(gpio_num_t::GPIO_NUM_16, gpio_num_t::GPIO_NUM_17, CAN_MODE_NORMAL);
-    g_config.mode = CAN_MODE_NORMAL;
-
-    can_timing_config_t t_config = CAN_TIMING_CONFIG_500KBITS();
-    can_filter_config_t f_config = {.acceptance_code = 0, .acceptance_mask = 0xFFFFFFFF, .single_filter = true};
-
-    //Filter out all messages except 0x305 and 0x307
-    //https://docs.espressif.com/projects/esp-idf/en/v3.3.5/api-reference/peripherals/can.html
-    //01100000101 00000 00000000 00000000 = 0x60A00000  (0x305)
-    //01100000111 00000 00000000 00000000 = 0x60E00000  (0x307)
-    //00000000010 11111 11111111 11111111 = 0x005FFFFF
-    //         ^ THIS BIT IS IGNORED USING THE MASK SO 0x305 and 0x307 are permitted
+    // Initialize configuration structures using macro initializers
+
+    twai_general_config_t g_config = TWAI_GENERAL_CONFIG_DEFAULT(gpio_num_t::GPIO_NUM_16, gpio_num_t::GPIO_NUM_17, TWAI_MODE_NORMAL);
+    g_config.mode = twai_mode_t::TWAI_MODE_NORMAL;
```

Comment:
this force override of mode can be dropped as it should be configured correctly by `TWAI_GENERAL_CONFIG_DEFAULT`.

Reply:
Thanks